jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/FilesTask.java

```diff
@@ -96,25 +97,35 @@ private static void printProjectFiles(Project project) {
   }
 
   /**
-   * Recursive function for printing out a project's artifacts. Calls itself when it encounters a
-   * project dependency.
+   * Function for collecting a project's project dependencies.
    *
-   * @param project the project to list the artifacts for
-   * @param projectDependenciesResult the set of project dependencies encountered. When a project
-   *     dependency is encountered, it is added to this set.
+   * @param project the project to find the project dependencies for
+   * @return the set of project dependencies
    */
-  private static void findProjectDependencies(
-      Project project, Set<ProjectDependency> projectDependenciesResult) {
-    for (Configuration configuration :
-        project.getConfigurations().getByName("runtime").getHierarchy()) {
-      for (Dependency dependency : configuration.getDependencies()) {
-        if (dependency instanceof ProjectDependency) {
-          projectDependenciesResult.add((ProjectDependency) dependency);
-          findProjectDependencies(
-              ((ProjectDependency) dependency).getDependencyProject(), projectDependenciesResult);
+  private static Set<ProjectDependency> findProjectDependencies(Project project) {
+    Set<ProjectDependency> projectDependencies = new HashSet<>();
+    ArrayDeque<Project> projects = new ArrayDeque<>();
```

Comment:
nit: Not entirely sure, but for a queue, we might want the type to be `Queue` (doesn't need to be double-ended here) and use `add` and `remove` (or `offer` and `poll`). `push` and `pop` are more for using `ArrayDeque` as a stack.

Reply:
That was intended, is there any reason we should use a queue instead of a stack here?

Reply:
Oh okay, that makes sense. Then maybe use the `Deque` type?